### PR TITLE
manifests/deployment: comply to restricted pod security level

### PR DESCRIPTION
**Pull request by s-urbaniak**
Starting from OpenShift 4.11 pod security admission is being activated. In OpenShift the default pod security admission level is going to be `restricted`. This PR fixes workloads from this repository. Concretely, the following violations have been detected:

```
{
  "objectRef": "openshift-marketplace/deployments/marketplace-operator",
  "pod-security.kubernetes.io/audit-violations": "would violate PodSecurity \"restricted:latest\": allowPrivilegeEscalation != false (container \"marketplace-operator\" must set securityContext.allowPrivilegeEscalation=false), unrestricted capab
ilities (container \"marketplace-operator\" must set securityContext.capabilities.drop=[\"ALL\"]), runAsNonRoot != true (pod or container \"marketplace-operator\" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container \"ma
rketplace-operator\" must set securityContext.seccompProfile.type to \"RuntimeDefault\" or \"Localhost\")"
}
```

/cc @stlaz

### CODE REVIEW

**Comment by timflannagan** [comment]
All the e2e related failures look relative to these changes.

/hold

**Comment by s-urbaniak** [comment]
Indeed, this is pending the changes that @stlaz is working on to align SCCs with pod security admission. Let's leave this open until that work is merged.

**Comment by s-urbaniak** [comment]
re-revert is now merged, this workload needs the runasuser UID removed such that it picks up the restricted-v2 SCC.

**Review by timflannagan — APPROVED** [review]

**Comment by timflannagan** [comment]
(whoops it looks like there's no approved-as-lgtm enabled for this repository)

/lgtm

**Pull request merged**
